src/App.js

```diff
@@ -40,16 +40,31 @@ class Board extends Component {
         this.onChangeDescription = this.onChangeDescription.bind(this);
     }
 
-    onOpenItemDetails(itemDetails) {
-        this.setState({ itemDetails, isDisplayedDetails: true })
+    onOpenItemDetails(itemDetails, id) {
+        this.setState({ 
+            itemDetails, 
+            isDisplayedDetails: true, 
+            selectedItemId: id
+        })
     }
 
     onCloseItemDetails() {
         this.setState({...initState})
     }
 
-    onChangeDescription(description) {
-        this.setState({itemDetails: {...this.state.itemDetails, description}})
+    onChangeDescription(newDescription) {
+
```

Comment:
I'm not sure I'm understanding what you mean with this comment, can you explain further?

Reply:
Oh, I see what you mean. I changed the name because it would be clearer when using the argument in the function, to properly differentiate between new and old description.